### DISCUSSION TITLE
Deployment as cloud function

### DISCUSSION THREAD

**Discussion by levino** [kimai/kimai#3616] [Questions & Answers]
Is it possible to deploy kimai as a google cloud / aws lambda function / serverless? Would be the easiest way to deploy it for a lot of people, I guess. Docker is fine, but usually requires a server, maybe even k8s if done right...

**Comment by kevinpapst** [comment]
> easiest way to deploy it for a lot of people, I guess.

That is your bubble, no one ever asked for that 😁 

> google cloud / aws lambda function

Lets throw money at US based tech-giants to host a free, one-man open source project. 

Well anyway ... I have no clue how these products work. But I am happy to assist to set it up and write documentation for it!

**Reply by levino** [reply]
Netlify for example allows you to host lambda functions for free with a pretty generous amount of executions per month, I think. So in theory one could host this service for free (excluding database fees). What is the alternative? I want to use your software to track time for parents work in our school. Not really a business, no money available. I want to keep cost as low as possible. What do you suggest? I guess this would run on some cheap webspace too? More or less like wordpress?

**Reply by levino** [reply]
And thank you for the quick reply, much appreciated.

**Reply by levino** [reply]
I meanwhile found the quite extensive and detailed docs for [recommended cheap deployments](https://www.kimai.org/documentation/installation.html#recommended-setup). I guess that part is answered. Sorry for raising the question.

**Reply by kevinpapst** [reply]
I am NOT a fan of shared hosting. They tend to make troubles due to old and weird setups.

Get your own server for 5€ / month: https://www.hetzner.com/de/cloud

I think that is rather cheap ... but has the downside that it requires a little bit of knowledge.

But this might help to set it up: https://www.kimai.org/documentation/fresh-ubuntu-20.html
